otherlibs/stdune/src/io.ml

```diff
@@ -252,23 +252,24 @@ struct
     (* Bindings to mac's fast copy function. It's similar to a hardlink, except
        it does COW when edited. It will also default back to regular copying if
        it fails for w/e reason *)
-    external copyfile : string -> string -> unit = "stdune_copyfile"
+    external copyfile : string -> string -> dst_maybe_present:bool -> unit
+      = "stdune_copyfile"
 
     external available : unit -> bool = "stdune_is_darwin"
   end
 
-  let copy_file ?chmod ~src ~dst () =
+  let gen_copy_file ?chmod ~src ~dst ~dst_maybe_present:_ () =
     Exn.protectx (setup_copy ?chmod ~src ~dst ()) ~finally:close_both
       ~f:(fun (ic, oc) -> copy_channels ic oc)
 
-  let copy_file =
+  let gen_copy_file =
     match Copyfile.available () with
-    | false -> copy_file
+    | false -> gen_copy_file
     | true -> (
-      fun ?chmod ~src ~dst () ->
+      fun ?chmod ~src ~dst ~dst_maybe_present () ->
         let src = Path.to_string src in
         let dst = Path.to_string dst in
-        (try Copyfile.copyfile src dst with
+        (try Copyfile.copyfile src dst ~dst_maybe_present with
```

Comment:
what's the goal of this labeled arg? It's always true for `Copyfile.copyfile` it seems?

Reply:
Yes, but it's not always true when `~atomically:true`. The goal is to shave off an `unlink` that we know is always going to be useless on macos.

Reply:
ooops, missed that. Thanks!